### PR TITLE
Updating routing doco to the most recent changes

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@SzymonPobiega this is minor feedback. But relying on the browser to wrap long lines in snippets after rendering can result in some hard to read code. Could u try to wrap the longer lines at a more sensible point in the cs files?

File: nservicebus/messaging/routing.md [hunk 1]
```diff
@@ -131,30 +131,29 @@ snippet:Routing-CustomRoutingStore
...
-#### Default mapping
-
-By default, NServiceBus assumes that each endpoint has a single non-scaled-out instance without a *discriminator*. This default allows for running without any configuration on transports like SQL Server or RabbitMQ.
+The purpose of the endpoint instance mapping layer is to provide information about physical deployments of a given endpoint. It is usually the responsibility of **operations engineers** who control environments and have the knowledge where a given endpoint has been deployed. Endpoint instance mapping is optional for most transports in most cases because of built-in conventions. An example of situation where it is required is a scaled out endpoint using MSMQ transport. Other endpoints who send messages to it need to be aware of its instances.
 
 
 #### Using config file
 
-In most scenarios endpoint mapping should be done via a config file so it can be modified without the need for re-deploying the binaries. 
+Usually endpoint mapping should be done via a config file so it can be modified without the need for re-deploying the binaries.
```
Comment (SimonCropp) [inline nservicebus/messaging/routing.md]: Usually is redundant when we have should

File: nservicebus/messaging/routing.md [hunk 2]
```diff
@@ -131,30 +131,29 @@ snippet:Routing-CustomRoutingStore
...
 #### Using config file
 
-In most scenarios endpoint mapping should be done via a config file so it can be modified without the need for re-deploying the binaries. 
+Usually endpoint mapping should be done via a config file so it can be modified without the need for re-deploying the binaries. 
 
 snippet:Routing-FileBased-Config
 
 To read more see [file-based endpoint mapping](/nservicebus/messaging/file-based-routing.md).
 
+When using transports other than MSMQ the filed-based mapping can be configured via *advanced* API:
```
Comment (SimonCropp) [inline nservicebus/messaging/routing.md]: Advanced is in the eye of the beholder. Just say "the folloing way" or something similar

File: nservicebus/messaging/routing.md [hunk 3]
```diff
@@ -131,30 +131,29 @@ snippet:Routing-CustomRoutingStore
...
+
+snippet:Routing-FileBased-ConfigAdvanced
+
 NOTE: If using a static type mapping to an address instead of an endpoint the advantages of file-based instance resolution will not be possible.
 
 
 #### Static mapping
 
-The `EndpointInstances` class provides a method that allows the registration of a static mapping. This API is useful in very specific scenarios where developers need to take control over a certain mapping.
+The `EndpointInstances` class provides a method that allows the registration of a static mapping. This API is useful in very rare and specific scenarios where developers need to take control over a certain mapping.
```
Comment (SimonCropp) [inline nservicebus/messaging/routing.md]: Very rare is kind of rendundant. If the scenario is adequately described then the rareness should be able to be inferred, or not, based on a given environment

File: nservicebus/messaging/routing.md [hunk 4]
```diff
@@ -131,30 +131,29 @@ snippet:Routing-CustomRoutingStore
...
-### Endpoint mapping layer
-
-Mapping of an endpoint to the collection of instances is the responsibility of **operations engineers**. They usually know where a given endpoint has been deployed.
-
+### Endpoint instance mapping layer
 
-#### Default mapping
-
-By default, NServiceBus assumes that each endpoint has a single non-scaled-out instance without a *discriminator*. This default allows for running without any configuration on transports like SQL Server or RabbitMQ.
+The purpose of the endpoint instance mapping layer is to provide information about physical deployments of a given endpoint. It is usually the responsibility of **operations engineers** who control environments and have the knowledge where a given endpoint has been deployed. Endpoint instance mapping is optional for most transports in most cases because of built-in conventions. An example of situation where it is required is a scaled out endpoint using MSMQ transport. Other endpoints who send messages to it need to be aware of its instances.
```
Comment (SimonCropp) [inline nservicebus/messaging/routing.md]: "Endpoints that send". "Who" sounds like they are sentient

Comment (SzymonPobiega) [inline nservicebus/messaging/routing.md]: :+1:

File: nservicebus/messaging/routing.md (hunk 2 above)
Comment (SzymonPobiega) [inline nservicebus/messaging/routing.md]: We actually call `config.UnicastRouting.Advanced.Blabla`. But you are right, the snippet itself shows that it is *advanced*.

Comment (SimonCropp) [inline nservicebus/messaging/routing.md]: @SzymonPobiega well the `config.UnicastRouting.Advanced` is wrong. there is no magic "this is advance and this isnt" it is a sliding scale. and often the worst people to decide "what is advanced" are the people writing the API.

**Comment by SzymonPobiega** [comment]
@SimonCropp Updated `Advanced` -> `Mapping`.

**Comment by weralabaj** [comment]
@SzymonPobiega is this WIP? or could be merged after review and rebase?

**Comment by SzymonPobiega** [comment]
It was a WIP because the changes were not merged yet. Now that Daniel merged the original PR, I'll update this one and will be ready to merge.

**Comment by weralabaj** [comment]
Ah ok, awesome :D

**Comment by SzymonPobiega** [comment]
Rebased and updated to latest core develop (1723).

File: nservicebus/messaging/routing.md [hunk 5]
```diff
@@ -78,7 +78,7 @@ Unicast routing uses queues which are inherently point-to-point channels. Each e
 
 ### Multicast routing
 
-Multicast routing uses topics. Multiple receivers can subscribe to a single topic and each message published to the topic is copied by the messaging infrastructure and delivered to each interested receiver.
+Multicast routing uses topics or similar devices. Multiple receivers can subscribe to a single topic and each message published to the topic is copied by the messaging infrastructure and delivered to each interested receiver.
```
Comment (weralabaj) [inline nservicebus/messaging/routing.md]: > or similar devices

Would it be more precise as:
"or equivalent features provided by the given transport"?

Comment (SzymonPobiega) [inline nservicebus/messaging/routing.md]: :+1:

Comment (weralabaj) [inline nservicebus/messaging/routing.md]: > Multiple receivers can subscribe to a single topic and each message published to the topic is copied by the messaging infrastructure and delivered to each interested receiver.

Too long:

"Multiple receivers can subscribe to a single topic. In such case every published message is copied by the messaging infrastructure, and the copy is delivered to all interested receivers."

Comment (SzymonPobiega) [inline nservicebus/messaging/routing.md]: :+1:

File: nservicebus/messaging/routing.md [hunk 6]
```diff
@@ -131,30 +131,29 @@ snippet:Routing-CustomRoutingStore
...
-### Endpoint mapping layer
-
-Mapping of an endpoint to the collection of instances is the responsibility of **operations engineers**. They usually know where a given endpoint has been deployed.
-
+### Endpoint instance mapping layer
 
-#### Default mapping
-
-By default, NServiceBus assumes that each endpoint has a single non-scaled-out instance without a *discriminator*. This default allows for running without any configuration on transports like SQL Server or RabbitMQ.
+The purpose of the endpoint instance mapping layer is to provide information about physical deployments of a given endpoint. It is usually the responsibility of **operations engineers** who control environments and have the knowledge where a given endpoint has been deployed. Endpoint instance mapping is optional for most transports in most cases because of built-in conventions. An example of situation where it is required is a scaled out endpoint using MSMQ transport. Other endpoints that send messages to it need to be aware of its instances.
```
Comment (weralabaj) [inline nservicebus/messaging/routing.md]: > It is usually the responsibility of **operations engineers** who control environments and have the knowledge where a given endpoint has been deployed. Endpoint in

You mean "Providing that information/configuration is usually..."?

**Comment by weralabaj** [comment]
I've added a few comments, apart from them IMO it's good to go.

File: nservicebus/messaging/routing.md (hunk 6 above)
Comment (SzymonPobiega) [inline nservicebus/messaging/routing.md]: Removed the whole sentence about ops (and also removed similar sentence regarding architects).

**Comment by SzymonPobiega** [comment]
@weralabaj I pushed new version.

**Comment by SzymonPobiega** [comment]
:tada:

**Pull request merged**